jamazon/main.js

```diff
@@ -175,17 +211,66 @@ function showHidden(view) {
   })
 }
 
+const $appCatalog = document.querySelector('[data-view="catalog"]')
+const $appDetails = document.querySelector('[data-view="details"]')
+const $cart = document.querySelector('[data-view="cart"]')
+const $appCart = document.querySelector('#cart')
+
+$appCatalog.addEventListener('click', function (element) {
+  const item = element.target.parentNode.closest('.card')
+  if (item) {
+    const itemId = item.getAttribute('data-item-id')
+    app.view = 'details'
+    app.details.item = app.catalog.items[itemId - 1]
+    catalog(app.catalog)
+    renderAll()
+  }
+})
+
+$appDetails.addEventListener('click', function (event) {
+  if (event.target.getAttribute('id') === 'addcart') {
+    app.cart.items.push(app.details.item)
+  }
+  renderAll()
+})
+
+$appDetails.addEventListener('click', function (event) {
+  if (event.target.getAttribute('id') === 'return') {
+    app.view = 'catalog'
+  }
+  renderAll()
+})
+
+$appCart.addEventListener('click', function (event) {
+  if (event.target.getAttribute('id') === 'cart-button') {
+    app.view = 'cart'
+  }
+  renderAll()
+})
+
+$cart.addEventListener('click', function (event) {
+  if (event.target.getAttribute('id') === 'returncart') {
+    app.view = 'catalog'
+  }
+  renderAll()
+})
+
 function renderAll() {
   const $appCatalog = document.querySelector('[data-view="catalog"]')
   const $appDetails = document.querySelector('[data-view="details"]')
+  const $cartSum = document.querySelector('[data-view="cart"]')
   const $cartLogo = document.querySelector('#cart')
-  if (app.view === 'catalog') {
+  if (app.view === 'catalog' && $appCatalog.innerHTML === '') {
```

Comment:
Instead of checking the `innerHTML`, just make it what you need, which is empty.